libfuse/folderlist.go

```diff
@@ -52,6 +56,26 @@ func (fl *FolderList) reportErr(ctx context.Context,
 	fl.fs.errLog.CDebugf(ctx, err.Error())
 }
 
+func (fl *FolderList) addToRecentlyRemove(name libkbfs.CanonicalTlfName) {
+	fl.muRecentlyRemoved.Lock()
+	if fl.recentlyRemoved == nil {
+		fl.recentlyRemoved = make(map[libkbfs.CanonicalTlfName]bool)
+	}
+	fl.recentlyRemoved[name] = true
+	fl.muRecentlyRemoved.Unlock()
```

Comment:
We find it easier to check for correctness when the Unlock happens in a `defer` right after the Lock.  In this case, I guess you don't want to hold the lock across the `execAfterDelay` call, so you can wrap this first part in an anonymous `func()` like:

```go
func() {
    fl.muRecentlyRemoved.Lock()
    defer fl.muRecentlyRemoved.Unlock()
    if fl.recentlyRemoved ...
    ...
}
fl.fs.execAfterDelay(time.Second, func() {
	fl.muRecentlyRemoved.Lock()
    defer fl.muRecentlyRemoved.Unlock()
	delete(fl.recentlyRemoved, name)
})
```